### ISSUE TITLE
del_atom problem in geometry.py

### ISSUE THREAD

**Issue by blakeo2** [SEDenmarkLab/molli#100]
The following code works fine from the `ml.Molecule`:
```
dendrobine = ml.Molecule.load_mol2(ml.files.dendrobine_mol2)
dendrobine.del_atom(0)
```

del_atom in `ml.CartesianGeometry` currently errors out with the following code
```
dendrobine = ml.Molecule.load_mol2(ml.files.dendrobine_mol2)
cartgeom = ml.CartesianGeometry(dendrobine)
cartgeom.del_atom(0)
>>>ValueError: list.remove(x): x not in list
```
This is caused by the fact that del_atom does not `get_atom` first.
This can be fixed by changing the del_atom from this:
```
def del_atom(self, _a: AtomLike) -> None:
    self._atoms.remove(_a)
```
to this:
```
def del_atom(self, _a: AtomLike) -> None:
    a = self.get_atom(_a)
    self._atoms.remove(a)
```

**Comment by esalx** [comment]
Fixed, added a mini-unit test to check for this in the future.

**Comment by esalx** [comment]
Closed with #141
